code/modules/mob/living/basic/lavaland/raptor/raptor.dm

```diff
@@ -0,0 +1,330 @@
+	icon_state = "raptor_yellow"
+	icon_living = "raptor_yellow"
+	icon_dead = "raptor_yellow_dead"
+	ridable_component = /datum/component/riding/creature/raptor/fast
+	speed = 1
+	raptor_color = RAPTOR_YELLOW
+	dex_description = "This breed possesses greasy fast speed, DEMON speed, making light work of long pilgrimages. It's said that a thunderclap could be heard when this breed reaches its maximum speed."
+	child_path = /mob/living/basic/mining/raptor/baby_raptor/yellow
+
+/mob/living/basic/mining/raptor/blue
+	name = "blue raptor"
+	icon_state = "raptor_blue"
+	icon_living = "raptor_blue"
+	icon_dead = "raptor_blue_dead"
+	raptor_color = RAPTOR_BLUE
+	dex_description = "Known to produce nutritous and equally delicious milk, which is also said to possess healing properties."
+	child_path = /mob/living/basic/mining/raptor/baby_raptor/blue
+
+/mob/living/basic/mining/raptor/blue/Initialize(mapload)
+	. = ..()
+	AddComponent(\
+		/datum/component/udder,\
+		udder_type = /obj/item/udder/raptor,\
+	)
+
+/obj/item/food/egg/raptor_egg
+	icon = 'icons/mob/simple/lavaland/raptor.dmi'
+	icon_state = "raptor_egg"
+	///inheritance datum to pass on to the child
+	var/datum/raptor_inheritance/inherited_stats
+
+/obj/item/food/egg/raptor_egg/proc/determine_growth_path(mob/living/basic/mining/raptor/dad, mob/living/basic/mining/raptor/mom)
+	if(dad.type == mom.type)
+		add_growth_component(dad.child_path)
+		return
+	var/dad_color = dad.raptor_color
+	var/mom_color = mom.raptor_color
+	var/list/my_colors = list(dad_color, mom_color)
+	sortTim(my_colors, GLOBAL_PROC_REF(cmp_text_asc))
+	for(var/path in GLOB.raptor_growth_paths) //guaranteed spawns
+		var/list/required_colors = GLOB.raptor_growth_paths[path]
+		if(!compare_list(my_colors, required_colors))
+			continue
+		add_growth_component(path)
+		return
+	var/list/valid_subtypes = list()
+	var/static/list/all_subtypes = subtypesof(/mob/living/basic/mining/raptor/baby_raptor)
+	for(var/path in all_subtypes)
+		var/mob/living/basic/mining/raptor/baby_raptor/raptor_path = path
+		if(!prob(initial(raptor_path.roll_rate)))
+			continue
+		valid_subtypes += raptor_path
+	add_growth_component(pick(valid_subtypes))
+
+/obj/item/food/egg/raptor_egg/proc/add_growth_component(growth_path)
+	if(GLOB.raptor_count >= MAX_RAPTOR_POP)
+		return
+	AddComponent(\
+		/datum/component/fertile_egg,\
+		embryo_type = growth_path,\
+		minimum_growth_rate = 0.5,\
+		maximum_growth_rate = 1,\
+		total_growth_required = 300,\
+		current_growth = 0,\
+		location_allowlist = typecacheof(list(/turf)),\
+		post_hatch = CALLBACK(src, PROC_REF(post_hatch)),\
+	)
+
+/obj/item/food/egg/raptor_egg/proc/post_hatch(mob/living/basic/mining/raptor/baby)
+	if(!istype(baby))
+		return
+	QDEL_NULL(baby.inherited_stats)
+	baby.inherited_stats = inherited_stats
+	inherited_stats = null
+
+/obj/item/food/egg/raptor_egg/Destroy()
+	QDEL_NULL(inherited_stats)
+	return ..()
+
+#define RANDOM_INHERIT_AMOUNT 2
+/datum/raptor_inheritance
```

Comment:
this file is already pretty big can you move this to another file (it can be in this folder)